src/hotspot/share/classfile/loaderConstraints.hpp

```diff
@@ -25,111 +25,47 @@
 #ifndef SHARE_CLASSFILE_LOADERCONSTRAINTS_HPP
 #define SHARE_CLASSFILE_LOADERCONSTRAINTS_HPP
 
-#include "classfile/placeholders.hpp"
-#include "utilities/hashtable.hpp"
+#include "oops/oopsHierarchy.hpp"
+#include "runtime/handles.hpp"
 
 class ClassLoaderData;
+class LoaderConstraint;
 class LoaderConstraintEntry;
```

Comment:
LoaderConstraintEntry should be removed.

Reply:
Thanks for spotting that.  I found we don't need to forward declare PlaceholderTable either.